crates/napi/src/threadsafe_function.rs

```diff
@@ -211,9 +211,13 @@ impl<T: 'static, ES: ErrorStrategy::T> ThreadsafeFunction<T, ES> {
       )
     })?;
 
+    let aborted = Arc::new(AtomicBool::new(false));
+    let aborted_ptr = Arc::into_raw(aborted.clone()) as *mut c_void;
+    check_status!(unsafe { sys::napi_add_env_cleanup_hook(env, Some(cleanup_cb), aborted_ptr) })?;
```

Comment:
An env cleanup hook is registered, and in the cleanup callback, aborted is set true so that ThreadsafeFunction will stop making use of the underlying tsfn handle